WordPress/src/test/java/org/wordpress/android/ui/sitecreation/MainVM/NewSiteCreationMainVMTest.kt

```diff
@@ -0,0 +1,209 @@
+import org.wordpress.android.ui.sitecreation.SiteCreationStep.SEGMENTS
+import org.wordpress.android.ui.sitecreation.SiteCreationStep.SITE_INFO
+import org.wordpress.android.ui.sitecreation.SiteCreationStep.SITE_PREVIEW
+import org.wordpress.android.ui.sitecreation.SiteCreationStep.VERTICALS
+import org.wordpress.android.ui.sitecreation.misc.NewSiteCreationTracker
+import org.wordpress.android.ui.sitecreation.previews.NewSitePreviewViewModel.CreateSiteState
+import org.wordpress.android.ui.sitecreation.previews.NewSitePreviewViewModel.CreateSiteState.SiteCreationCompleted
+
+private const val LOCAL_SITE_ID = 1
+private const val SEGMENT_ID = 1L
+private const val VERTICAL_ID = "m1v1"
+private const val SITE_TITLE = "test title"
+private const val SITE_TAGLINE = "test tagline"
+private const val DOMAIN = "test.domain.com"
+
+private val SITE_CREATION_STEPS = listOf(SITE_INFO, VERTICALS, SEGMENTS, DOMAINS, SITE_PREVIEW)
+
+@RunWith(MockitoJUnitRunner::class)
+class NewSiteCreationMainVMTest {
+    @Rule
+    @JvmField val rule = InstantTaskExecutorRule()
+
+    @Mock lateinit var tracker: NewSiteCreationTracker
+    @Mock lateinit var stepsProvider: NewSiteCreationStepsProvider
+    @Mock lateinit var navigationTargetObserver: Observer<NavigationTarget>
+    @Mock lateinit var wizardFinishedObserver: Observer<CreateSiteState>
+    @Mock lateinit var savedInstanceState: Bundle
+
+    @Captor
+    private val captorNavigationTarget: ArgumentCaptor<NavigationTarget>? = null
+
+    private lateinit var viewModel: NewSiteCreationMainVM
+
+    @Before
+    fun setUp() {
+        whenever(stepsProvider.getSteps()).thenReturn(SITE_CREATION_STEPS)
+        viewModel = NewSiteCreationMainVM(stepsProvider, tracker)
+        viewModel.start(null)
+        viewModel.navigationTargetObservable.observeForever(navigationTargetObserver)
+        viewModel.wizardFinishedObservable.observeForever(wizardFinishedObserver)
+    }
+
+    @Test
+    fun skipClickedResultsInNextStep() {
+        viewModel.onSkipClicked()
+        verifyNavigatedToNextStep()
+    }
+
+    @Test
+    fun segmentSelectedResultsInNextStep() {
+        viewModel.onSegmentSelected(SEGMENT_ID)
+        verifyNavigatedToNextStep()
+    }
+
+    @Test
+    fun verticalSelectedResultsInNextStep() {
+        viewModel.onVerticalsScreenFinished(VERTICAL_ID)
+        verifyNavigatedToNextStep()
+    }
+
+    @Test
+    fun siteInfoFinishedResultsInNextStep() {
+        viewModel.onInfoScreenFinished(SITE_TITLE, null)
+        verifyNavigatedToNextStep()
+    }
+
+    @Test
+    fun domainSelectedResultsInNextStep() {
+        viewModel.onDomainsScreenFinished(DOMAIN)
+        verifyNavigatedToNextStep()
+    }
+
+    @Test
+    fun siteCreationStateUpdatedWithSelectedSegment() {
+        viewModel.onSegmentSelected(SEGMENT_ID)
+        assertThat(viewModel.navigationTargetObservable.lastEvent!!.wizardState.segmentId).isEqualTo(SEGMENT_ID)
```

Comment:
Nitpick: It might be worth adding a helper for `viewModel.navigationTargetObservable.lastEvent!!.wizardState` as it's used fairly often.